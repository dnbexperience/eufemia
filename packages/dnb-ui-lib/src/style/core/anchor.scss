/*
 * Anchor
 *
 * ========================================================================== */

@mixin hoverStyle() {
  background-color: var(--color-mint-green-50);
  border-bottom-width: 2px;
}
@mixin activeStyle() {
  color: #fff;
  background-color: var(--color-sea-green);
  border-bottom: none;
}
@mixin focusStyle($whatinput: null) {
  @include fakeFocus($whatinput);
  border-bottom: none;
}

a {
  // make sure we have always `display: inline;` as inline-block is breaking the border-bottom
  display: inline;

  // top has 1px less, cause we now has 2px on border bottom hover
  // the calculations are based on 24px line-height

  // 1px = 0,0625
  // 2px = 0,125
  // 4px = 0,25

  // on padding-top we have 4px
  // on padding-bottom we have 2px, cause run a border-bottom with 2px
  // this makes a total of 8px
  // 24px line-height - 8px = 16px (font-size)
  padding: 0.25em 0.125em 0.125em;

  // margins are not needed as long as we go for "inline"
  // margin: -0.25em 0 -0.125em;

  color: inherit;
  color: var(--color-sea-green, inherit);
  text-decoration: none;
  border-bottom: 1px solid var(--color-sea-green);

  &:active,
  &:hover {
    @include hoverStyle();

    // margins are not needed as long as we go for "inline"
    // margin-bottom: -0.1875em;
  }
  &:active {
    @include activeStyle();
  }
  &:focus {
    @include fakeFocus();
  }

  transition: background-color 0.2s ease;

  // other stuff, not related to the Anchor directly
  .dnb-icon {
    display: inline-block;
    // -1px to center the icons agains large Chars
    transform: translateY(-1px);
  }
}
.dnb-hover-style {
  @include hoverStyle();
}
.dnb-active-style {
  @include activeStyle();
}
.dnb-focus-style {
  @include focusStyle('mouse');
}
.dnb-with-animation {
  position: relative; // because of z-index and underline/hover sudo classes
  z-index: 1; // because of focus handling

  // This way we make sure we never have a line break in an Anchor
  // Cause this messes up the "drawUnderlineBorder" mechanism
  white-space: pre;

  // border
  @include drawUnderlineBorder();

  // bg
  @include drawUnderlineBackground();

  background-color: transparent;
  border-bottom: none;

  &:active,
  &:hover {
    background-color: transparent;
    border-bottom: none;

    // bg
    &::before {
      max-height: 100%;
    }

    // border
    &::after {
      left: 0.1em;
      right: 0.1em;
      border-bottom-width: 2px;
    }
  }
  &:active {
    border-bottom: none;

    // bg
    &::before {
      background-color: var(--color-sea-green);
    }
    // border
    &::after {
      border-bottom-color: var(--color-sea-green);
    }
  }
  &:focus {
    @include fakeFocus();
    // border
    &::after {
      border-bottom: none;
    }
  }
}

// resettings
@mixin resetUnderlineStyle() {
  border-bottom: none;
  // border
  &::after {
    border-bottom: none;
    content: none;
  }
}
@mixin resetHoverStyle() {
  &:hover {
    color: inherit;
    background-color: inherit;
    border-bottom-width: 1px;
  }

  // bg
  &::before {
    content: none;
  }
}
@mixin resetAnimationStyle() {
  transition: none;

  // bg
  &::before {
    transition: max-height 0.01s linear;
  }
}
.dnb-no-anchor-style {
  padding: 0;
  margin: 0;
  white-space: normal;
  border-bottom: none;
  color: inherit;

  // border
  &::after {
    content: none;
  }
  // bg
  &::before {
    content: none;
  }

  @include resetUnderlineStyle();
  @include resetHoverStyle();
}
.dnb-no-anchor-underline {
  @include resetUnderlineStyle();
}
.dnb-no-anchor-hover {
  @include resetHoverStyle();
}
.dnb-no-anchor-animation {
  @include resetAnimationStyle();
}
a:not([href]) {
  cursor: pointer;
  user-select: none;
}
