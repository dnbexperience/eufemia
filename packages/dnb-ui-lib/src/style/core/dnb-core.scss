/*
 * Core Styles loader
 *
 */

// core imports
@import './reset.scss';
@import './properties.scss';
@import './utilities.scss';
@import './anchor.scss';
@import './typography.scss';
@import './scopes.scss';

// Global body style
body {
  @include bodyDefault();
  @media print {
    background-color: white;
  }
}

// Focus management
html[data-whatinput='mouse'] .dnb-style *:focus {
  outline: none;
}
html[data-whatinput='keyboard'] .dnb-style *:not(a):not(div):focus {
  @include fakeFocus();
}
.dnb-no-focus {
  outline: none;
}
.dnb-fake-focus {
  @include fakeFocus();
}
.dnb-tab-focus {
  outline: none;
  &:focus {
    @include fakeFocus('keyboard');
  }
}
.dnb-always-focus {
  &:focus {
    @include fakeBorder(var(--color-signal-orange), null, !important);
  }
}

// Layout management
// Use this class where components are trying to add background boxes/borders/lines
// to the very outer sides of the page
// e.g. Tabs are using this technique
// In some cases we dont want to have that effect.
// E.g. if we have a menu to the left or right side
.dnb-page-content-inner {
  overflow: hidden;
}

// TODO: considder to take this out

$breakpoints: (
  'small': 320px,
  'medium': 700px,
  'large': 1024px,
  'xLarge': 1400px
);

.dnb-hide-on-mobile {
  @include allBelow(medium) {
    display: none;
  }
}
.dnb-mobile-exclusive {
  @include allAbove(medium) {
    display: none;
  }
}
.dnb-width-limit {
  margin: 0 auto;
  width: 100%;
  max-width: map-get($map: $breakpoints, $key: large);
  padding: 0 0.5rem;

  @include allAbove(medium) {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}
.dnb-belt {
  padding: 1rem 0;
  width: 100%;
  background: var(--color-sea-green-4);
}
.dnb-nudge {
  display: inline-flex;
  flex: 0 0 100%;

  @for $i from 1 through 10 {
    &--vertical {
      &[data-nudges='#{$i}'] {
        @include nudger(vertical, #{$i});
      }
    }
    &--horizontal {
      &[data-nudges='#{$i}'] {
        @include nudger(horizontal, #{$i});
      }
    }
  }
}
.dnb-sr-only {
  @include sr-only();
}
.dnb-not-sr-only {
  @include notSrOnly();
}
.dnb-unstyled-list {
  @include unstyledList();
}
