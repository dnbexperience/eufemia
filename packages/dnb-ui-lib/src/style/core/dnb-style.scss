/*
 * Core Styles
 * within the scope of ".dnb-style"
 *
 */

// core imports
// importing properties and utilities is not needed anymore
// @import './properties.scss';
// @import './utilities.scss';

// hey have to be used by a parent css class: .dnb-style
.dnb-style {
  small {
    font-size: 0.875rem; // 14px
    line-height: 1.5rem;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 1rem 0;
    padding: 0;

    font-family: var(--font-family-demi);
    line-height: 1rem; // 16px
    font-weight: 400;

    color: var(--color-emerald-green);
  }

  h1 {
    margin: 1.5rem 0 1rem;

    font-size: 3rem; // 48px
    line-height: 3.5rem; // 56px
    font-weight: 400;
  }

  h1.small,
  h1 small {
    display: block;
    font-size: 2rem; // 32px
    line-height: 3rem; // 48px
  }

  h2 {
    margin: 1.5rem 0 1rem;

    font-size: 1.5rem; // 24px
    line-height: 2rem; // 32px
  }

  /*
  * H3 looks the same as .dnb-lead
  *
  */
  h3,
  p.dnb-lead {
    font-family: var(--font-family-demi);
    font-size: 1.25rem; // 20px
    line-height: 2rem; // 32px
    font-weight: 400;
  }
  h3 > small,
  h3.small,
  p.dnb-lead > small,
  p.dnb-lead > .small {
    font-size: 1rem; // 16px
    line-height: 1.5rem; // 24px
  }

  // The rest of the Headings are nod defned by the Eufemia Typography
  h4 {
    font-size: 1.25rem; // 20px
    line-height: 1.5rem; // 24px
  }
  h5 {
    font-size: 1rem; // 16px
    line-height: 1.5rem; // 24px
  }
  h6 {
    font-size: 0.875rem;
  }

  p {
    // if we not reset margin, the browser is using: margin-block-end: 1rem;
    margin-top: 0;
    margin-bottom: 0;
    padding: 0;

    font-size: 1rem; // 16px
    line-height: 1.5rem; // 24px

    b,
    strong {
      font-weight: 600;
    }
  }

  label,
  input {
    font-size: 1rem; // 16px
  }
  form,
  fieldset {
    font-size: 1rem; // 16px
  }

  table {
    display: block;
    width: 100%;

    overflow: auto;
    margin-top: 0;
    margin-bottom: 1rem;

    border-spacing: 0;
    border-collapse: collapse;
  }
  table th,
  table td {
    white-space: nowrap;

    // to make sure we have the ability to have a border where ever we want
    border-bottom: 1px solid transparent;

    font-size: 1em;
    line-height: 1.5em;
  }
  table th {
    padding: 3em 1em 0.5em;

    font-weight: 600;
    text-align: left;
    color: var(--color-emerald-green);

    background-color: var(--color-sea-green-4);
    border-bottom: 1px solid var(--color-mint-green);
  }
  table td {
    padding: 1.0625em 1em;
    border-bottom: 1px solid transparent;

    line-height: 1.5em;
  }
  table tr {
    background-color: #fff;
  }
  table tr:nth-child(2n) {
    background-color: var(--color-sea-green-4);
  }

  // General Layout and Spacing + some Styling
  *:not([class^='dnb-']) + {
    p,
    blockquote,
    ul,
    ol,
    dl,
    pre {
      margin-top: 0;
      margin-bottom: 1rem;
    }

    ul,
    ol {
      padding-left: 2em;
      margin-top: 0;
      margin-bottom: 0;
      line-height: 1.5rem;
    }

    li {
      word-wrap: break-all;
    }

    ul li,
    ol li {
      margin-top: 1em;
      margin-bottom: 1em;
    }

    li > p {
      margin-top: 1em;
    }

    li + li {
      margin-top: 0.25em;
    }

    ol ol,
    ul ol {
      list-style-type: lower-roman;
    }

    ul ul,
    ul ol,
    ol ol,
    ol ul {
      margin-top: 0;
      margin-bottom: 0;
    }

    ul ul ol,
    ul ol ol,
    ol ul ol,
    ol ol ol {
      list-style-type: lower-alpha;
    }

    dd {
      margin-left: 0;
    }

    dl {
      padding: 0;
    }

    dl dt {
      padding: 0;
      margin-top: 1em;
      font-size: 1em;
      font-style: italic;
      font-weight: 600;
    }

    dl dd {
      padding: 0 1em;
      margin-bottom: 1em;
    }

    blockquote {
      padding: 0 1em;
      color: var(--color-summer-green);
      border-left: 4px dotted var(--color-summer-green);
    }

    blockquote > :first-child {
      margin-top: 0;
    }

    blockquote > :last-child {
      margin-bottom: 0;
    }

    img {
      box-sizing: content-box;
      max-width: 100%;
      border-style: none;
      background-color: #fff;
    }

    img[align='right'] {
      padding-left: 1.2rem;
    }

    img[align='left'] {
      padding-right: 1.2rem;
    }

    code,
    * code {
      margin: 0;
      padding: 0.2rem 0.4rem;

      font-size: 0.875em;
      background-color: var(--color-mint-green-25);
      border-radius: 0.1875em;
    }
    pre,
    * pre {
      word-wrap: normal;
    }
  }
}

.dnb-style-selection ::selection,
.dnb-style ::selection {
  background: var(--color-mint-green);
  color: var(--color-black);
  text-shadow: none;
}
