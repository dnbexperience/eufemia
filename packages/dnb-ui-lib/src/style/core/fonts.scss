/*
 * Typography
 *
 * The font-family properties are defined in the properties.scss file, like:
 * --font-family-default:
 *
 */

// Font usage
.dnb-typo-std,
.dnb-typo-book {
  font-family: var(--font-family-default);
  font-weight: normal;
  font-style: normal;
}
.dnb-typo-medium {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-medium);
  font-style: normal;
}
.dnb-typo-demi {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-demi);
  font-style: normal;
}

// Number usage
.dnb-typo-number {
  // Proportional Lining (default)
  &--lining {
    @include numberFeature(lining-nums, lnum);
  }
  // Proportional Tabular
  &--tabular {
    @include numberFeature(tabular-nums, lnum);
  }

  // Deprecated
  &--oldstyle {
    @include numberFeature(oldstyle-nums, onum);
  }
  // Deprecated
  &--proportional {
    @include numberFeature(proportional-nums, pnum);
  }
}

$fonts-path: '../../assets/fonts/' !default;

// Std and Book
@font-face {
  font-family: 'Fedra Sans Std';
  src: url('#{$fonts-path}/FedraSansStd-Book.woff2') format('woff2'),
    url('#{$fonts-path}/FedraSansStd-Book.woff') format('woff'),
    url('#{$fonts-path}/FedraSansStd-Book.ttf') format('truetype');
  font-weight: normal; // have to be the same as: --font-weight-book
  font-style: normal;
  font-display: fallback;
}

// Demi
@font-face {
  font-family: 'Fedra Sans Std';
  src: url('#{$fonts-path}/FedraSansStd-Demi.woff2') format('woff2'),
    url('#{$fonts-path}/FedraSansStd-Demi.woff') format('woff'),
    url('#{$fonts-path}/FedraSansStd-Demi.ttf') format('truetype');
  font-weight: 500; // have to be the same as: --font-weight-demi
  font-style: normal;
  font-display: fallback;
}

// Medium
@font-face {
  font-family: 'Fedra Sans Std';
  src: url('#{$fonts-path}/FedraSansStd-Medium.woff2') format('woff2'),
    url('#{$fonts-path}/FedraSansStd-Medium.woff') format('woff'),
    url('#{$fonts-path}/FedraSansStd-Medium.ttf') format('truetype');
  font-weight: 600; // have to be the same as: --font-weight-medium
  font-style: normal;
  font-display: fallback;
}
