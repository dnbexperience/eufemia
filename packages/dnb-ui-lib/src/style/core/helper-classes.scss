/*
 * Helper Classes
 *
 */

// Focus management
.dnb-no-focus {
  outline: none;
}
.dnb-tab-focus {
  outline: none;
  &:focus {
    @include fakeFocus('keyboard');
  }
}
.dnb-mouse-focus {
  outline: none;
  &:focus {
    @include fakeFocus('mouse');
  }
}
.dnb-focus-ring {
  @include fakeBorder($focusRingColor, $focusRingWidth, null, !important);
}
@import './helper-classes/skip-link.scss';

// Cosmetic helpers
@import './helper-classes/dnb-section.scss';
.dnb-unstyled-list {
  @include unstyledList();
}

// Layout management
// Use this class where components are trying to add background boxes/borders/lines
// to the very outer sides of the page
// e.g. Tabs are using this technique
// In some cases we dont want to have that effect.
// E.g. if we have a menu to the left or right side
.dnb-app-content-inner {
  overflow: hidden;
}
.dnb-hide-on-mobile {
  @include allBelow(medium) {
    display: none;
  }
}
.dnb-mobile-exclusive {
  @include allAbove(medium) {
    display: none;
  }
}
.dnb-width-limit {
  margin: 0 auto;
  width: 100%;
  max-width: map-get($map: $breakpoints, $key: large);
  padding: 0 0.5rem;

  @include allAbove(medium) {
    padding-left: 1rem;
    padding-right: 1rem;
  }
}
.dnb-belt {
  padding: 1rem 0;
  width: 100%;
  background: var(--color-mint-green-12);
}
.dnb-nudge {
  display: inline-flex;
  flex: 0 0 100%;

  @for $i from 1 through 10 {
    &--vertical {
      &[data-nudges='#{$i}'] {
        @include nudger(vertical, #{$i});
      }
    }
    &--horizontal {
      &[data-nudges='#{$i}'] {
        @include nudger(horizontal, #{$i});
      }
    }
  }
}
.dnb-sr-only {
  @include srOnly();
}
.dnb-not-sr-only {
  @include notSrOnly();
}

// Form
.dnb-form-group {
  display: inline-flex;
  flex-direction: row;

  width: 100%;

  > .dnb-form-status {
    margin: 0.5rem 0;
  }

  // for vertical labeling
  &__position--vertical {
    flex-direction: column;

    width: auto;

    > label,
    > .dnb-form-label {
      margin: 0;
      margin-bottom: 0.5rem;
    }
  }

  // for hoizontal labeling
  &,
  &__position--hoizontal {
    > label,
    > .dnb-form-label {
      margin: 0;
      margin-right: 1rem;
    }
  }
}
.dnb-form-row {
  margin-bottom: 1.5rem;
}
