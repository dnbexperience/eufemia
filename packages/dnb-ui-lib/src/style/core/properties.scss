/*
 * CSS Properties
 *
 */

:root {
  // Typography Family
  --font-family-std: 'Fedra Sans Std', sans-serif;
  --font-family-book: 'Fedra Sans Std', sans-serif; /* Do not use, use --font-weight-... */
  --font-family-demi: 'Fedra Sans Std', sans-serif; /* Do not use, use --font-weight-... */
  --font-family-medium: 'Fedra Sans Std', sans-serif; /* Do not use, use --font-weight-... */

  // Typography Weights
  --font-weight-std: normal;
  --font-weight-book: normal;
  --font-weight-demi: 500;
  --font-weight-medium: 600;

  // UX colors
  --color-sea-green-alt: #008484;
  --color-mint-green-50: #d2f0e9;
  --color-mint-green-25: #e9f8f4;
  --color-sea-green-4: #f5fafa;
  --color-sea-green-alt-30: #b3dada;
  --color-signal-yellow-30: #ffffd7;
  --color-cherry-red: #e10076;
  --color-cherry-red-80: #fdebf4;
  --color-black: #000;
  --color-black-80: #333;
  --color-black-30: #b3b3b3;
  --color-white: #fff;

  // Undefined
  --color-black-border: #cdcdcd;
  --color-black-background: #fafafa;

  // Profile colors
  --color-sea-green: #007272;
  --color-mint-green: #a5e1d2;
  --color-summer-green: #28b482;
  --color-emerald-green: #14555a;
  --color-ocean-green: #00343e;
  --color-signal-orange: #ff5400;
  --color-signal-yellow: #ffff7a;
  --color-indigo: #23195a;
  --color-violet: #6e2382;
  --color-sky-blue: #4bbed2;

  // Unused Profile colors
  // --color-mint-green-alt: #ebfffa;
  // --color-indigo-medium: #6e6491;
  // --color-indigo-light: #b9afc8;
  // --color-violet-medium: #a06eaf;
  // --color-violet-light: #cfb9d7;
  // --color-sky-blue-medium: #87d2e1;
  // --color-sky-blue-light: #c3ebf0;

  // Colors used in mobile
  // --color-light-grey: #fafafa;
  // --color-black-border: #ebebeb;
  // + new grey definitions
  // --color-subtle-grey: #f8f8f8;
  // --color-soft-grey: #ccc;
  // --color-charcoal-grey: #333;
  // --color-shadow-grey: #757575;
}
