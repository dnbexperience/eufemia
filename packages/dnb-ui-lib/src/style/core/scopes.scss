/*
 * Scopes
 *
 */

@import './reset.scss';

@mixin coreDefault() {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: var(--font-size-basis);
  font-style: normal;
  line-height: var(--line-height-basis);
  color: var(--color-black-80, #333);

  // The new DNB font needs font smoothing to be thinner on webkit and FF
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;

  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);

  @include numberFeature(lining-nums, lnum);

  @content;
}

@mixin bodyDefault() {
  @include reset();
  @include coreDefault();

  @media print {
    background-color: white;
  }

  @content;
}

@mixin htmlDefault() {
  html {
    font-size: 100%;
    @supports (font: -apple-system-body) {
      font: -apple-system-body; /* stylelint-disable-line */
    }
    scroll-behavior: smooth;
  }

  // reset ePlatform css
  @include resetLegacyStyles();
}

@mixin componentReset() {
  @include coreDefault();
  @include coreReset();

  margin: 0;
  padding: 0;

  @content;
}
