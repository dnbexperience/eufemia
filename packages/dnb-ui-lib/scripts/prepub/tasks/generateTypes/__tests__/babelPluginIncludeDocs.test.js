/**
 * Scripts test
 *
 */

import nodePath from 'path'
import { transformFileAsync } from '@babel/core'
import { babelPluginConfigDefaults } from '../babelPluginConfigDefaults'
import { babelPluginPropTypesRelations } from '../babelPluginPropTypesRelations'
import { fetchPropertiesFromDocs } from '../fetchPropertiesFromDocs'
import { babelPluginIncludeDocs } from '../babelPluginIncludeDocs'
import { babelPluginCorrectTypes } from '../babelPluginCorrectTypes'

describe('babelPluginIncludeDocs', () => {
  const docsDir = nodePath.resolve(__dirname, '__mocks__')
  const file = nodePath.resolve(docsDir, 'PrimaryComponent.js')
  let docs = {}

  async function runIncludeDocsTestSuite({ sourceDir, ...options }) {
    const { code } = await transformFileAsync(file, {
      plugins: [
        ['@babel/plugin-proposal-class-properties', { loose: true }],
        [babelPluginPropTypesRelations, { sourceDir }],
        [
          babelPluginCorrectTypes,
          {
            strictMode: false,
            ...options
          }
        ],
        [
          babelPluginIncludeDocs,
          {
            docs
          }
        ]
      ],
      ...babelPluginConfigDefaults
    })
    return code
  }

  it('has to match docs snapshot', async () => {
    docs = await fetchPropertiesFromDocs({
      file,
      docsDir,
      findFiles: ['PrimaryComponent.md']
    })
    expect(docs).toMatchSnapshot()
  })

  it('has to match code snapshot', async () => {
    const sourceDir = nodePath.dirname(file)
    const code = await runIncludeDocsTestSuite({ sourceDir })

    expect(code).toMatchSnapshot()
  })

  it('has to match code snapshot in strict mode', async () => {
    const sourceDir = nodePath.dirname(file)
    const code = await runIncludeDocsTestSuite({
      sourceDir,

      /**
       * If strictMode is enabled,
       * it will transform "string + bool" or "string + number" in to string or bool only
       */
      strictMode: true
    })

    expect(code).toMatchSnapshot()
  })
})
