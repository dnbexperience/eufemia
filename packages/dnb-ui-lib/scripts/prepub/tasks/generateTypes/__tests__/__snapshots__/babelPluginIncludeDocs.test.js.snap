// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelPluginIncludeDocs has to match code snapshot 1`] = `
"/**
 * Test mock file
 *
 */
import React from 'react';
import { PropTypes } from \\"react\\";
import SecondaryComponent, { defaultProps as secondaryDefaultProps, propTypes as secondaryPropTypes } from './SecondaryComponent';
export const primaryPropTypes = {
  top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
};
export const primaryDefaultProps = {
  space: null,
  top: null,
  right: null,
  bottom: null,
  left: null
};
export default class PrimaryComponent extends React.PureComponent {
  render() {
    return [this.props.boolean, this.props.number, this.props.spacing, this.props.foo, this.props.secondary_foo, this.props.secondary_spacing, this.props.children];
  }

}
PrimaryComponent.propTypes = {
  boolean: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  number: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),

  /**
   * Text \`spacing\` or \`children\`.
  */
  spacing: PropTypes.shape({
    top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),
  foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-top\`.
  */
  top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-right\`.
  */
  right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-bottom\`.
  */
  bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-left\`.
  */
  left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`secondary\`.
  */
  secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_spacing: PropTypes.shape({
    secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),

  /**
   * Text \`spacing\` or \`children\`.
  */
  children: PropTypes.node
};
PrimaryComponent.defaultProps = {
  boolean: null,
  number: null,
  spacing: null,
  foo: null,
  ...primaryDefaultProps,
  ...secondaryDefaultProps,
  children: null
};
PrimaryComponent.Secondary = SecondaryComponent;

const Element = () => {
  return null;
};

Element.propTypes = {
  boolean: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  number: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),

  /**
   * Text \`spacing\` or \`children\`.
  */
  spacing: PropTypes.shape({
    top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),
  foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-top\`.
  */
  top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-right\`.
  */
  right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-bottom\`.
  */
  bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-left\`.
  */
  left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`secondary\`.
  */
  secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_spacing: PropTypes.shape({
    secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),

  /**
   * Text \`spacing\` or \`children\`.
  */
  children: PropTypes.node
};
Element.defaultProps = {
  boolean: null,
  number: null,
  spacing: null,
  foo: null,
  ...primaryDefaultProps,
  children: null
};"
`;

exports[`babelPluginIncludeDocs has to match code snapshot in strict mode 1`] = `
"/**
 * Test mock file
 *
 */
import React from 'react';
import { PropTypes } from \\"react\\";
import SecondaryComponent, { defaultProps as secondaryDefaultProps, propTypes as secondaryPropTypes } from './SecondaryComponent';
export const primaryPropTypes = {
  top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
};
export const primaryDefaultProps = {
  space: null,
  top: null,
  right: null,
  bottom: null,
  left: null
};
export default class PrimaryComponent extends React.PureComponent {
  render() {
    return [this.props.boolean, this.props.number, this.props.spacing, this.props.foo, this.props.secondary_foo, this.props.secondary_spacing, this.props.children];
  }

}
PrimaryComponent.propTypes = {
  boolean: PropTypes.bool,
  number: PropTypes.number,

  /**
   * Text \`spacing\` or \`children\`.
  */
  spacing: PropTypes.shape({
    top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),
  foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-top\`.
  */
  top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-right\`.
  */
  right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-bottom\`.
  */
  bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-left\`.
  */
  left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`secondary\`.
  */
  secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_spacing: PropTypes.shape({
    secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),

  /**
   * Text \`spacing\` or \`children\`.
  */
  children: PropTypes.node
};
PrimaryComponent.defaultProps = {
  boolean: null,
  number: null,
  spacing: null,
  foo: null,
  ...primaryDefaultProps,
  ...secondaryDefaultProps,
  children: null
};
PrimaryComponent.Secondary = SecondaryComponent;

const Element = () => {
  return null;
};

Element.propTypes = {
  boolean: PropTypes.bool,
  number: PropTypes.number,

  /**
   * Text \`spacing\` or \`children\`.
  */
  spacing: PropTypes.shape({
    top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),
  foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-top\`.
  */
  top: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-right\`.
  */
  right: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-bottom\`.
  */
  bottom: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`margin-left\`.
  */
  left: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),

  /**
   * Text \`secondary\`.
  */
  secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_foo: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
  secondary_spacing: PropTypes.shape({
    secondary: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
  }),

  /**
   * Text \`spacing\` or \`children\`.
  */
  children: PropTypes.node
};
Element.defaultProps = {
  boolean: null,
  number: null,
  spacing: null,
  foo: null,
  ...primaryDefaultProps,
  children: null
};"
`;

exports[`babelPluginIncludeDocs has to match docs snapshot 1`] = `
Array [
  Object {
    "bottom": "Text \`margin-bottom\`.",
    "children": "Text \`spacing\` or \`children\`.",
    "left": "Text \`margin-left\`.",
    "right": "Text \`margin-right\`.",
    "secondary": "Text \`secondary\`.",
    "spacing": "Text \`spacing\` or \`children\`.",
    "top": "Text \`margin-top\`.",
  },
]
`;
